dist/gcc-compatible/EverCrypt_Hash.c

```diff
@@ -1450,7 +1450,7 @@ Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____
   *buf = (uint8_t *)KRML_HOST_CALLOC(EverCrypt_Hash_Incremental_block_len(a), sizeof (uint8_t));
   EverCrypt_Hash_state_s *block_state = EverCrypt_Hash_create_in(a);
   Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____
-  s = { .block_state = block_state, .buf = buf, .total_len = (uint64_t)0U };
+  s = { .block_state = block_state, .buf = buf, .total_len = (uint64_t)(uint32_t)0U };
```

Comment:
This double cast (and all the following ones, which probably come from the same root cause) is a bit unfortunate.

Reply:
don't think this is a big deal